Framework/Algorithms/src/LineProfile.cpp

```diff
@@ -87,6 +89,39 @@ struct IndexLimits {
   size_t widthEnd;
 };
 
+/**
+ * Create the profile workspace.
+ * @param ws The parent workspace.
+ * @param Xs Profile's X values.
+ * @param Ys Profile's Y values.
+ * @param Es Profile's E values.
+ * @return A single histogram profile workspace.
+ */
+Workspace2D_sptr makeOutput(const MatrixWorkspace &ws,
+                            const LineDirection direction,
+                            std::vector<double> &&Xs, std::vector<double> &&Ys,
+                            std::vector<double> &&Es) {
```

Comment:
Why use an rvalue reference if you are not moving from it below?

Reply:
`BinEdges` and `VectorOf` have move constructors. Do I still need to `move` manually? I'd thought `move` only casts to rvalue if it is possible.

Reply:
I think so, have a look: https://stackoverflow.com/questions/31213539/c-stdmove-with-rvalue-reference-is-not-moving-contents.